src/lib/renderElement.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  createElement__v2,
+  removeElementAttribute,
+  setElementAttributes,
+} from './createElement__v2';
+import {
+  isVNodeElement,
+  isVNodeText,
+  VNode,
+  VNodeChild,
+  VNodeElement,
+  VNodeProps,
+} from './createVNode';
+import { setupEventListeners } from './eventManager';
+
+/**
+ * 주어진 가상 DOM 노드(vNode)를 처리하여 렌더링 가능한 형태로 변환합니다.
+ *
+ * 이 함수는 다음을 수행합니다:
+ * - null, undefined, boolean 값을 빈 문자열로 처리
+ * - 문자열과 숫자를 문자열로 변환
+ * - 함수형 컴포넌트를 호출하고, 그 결과를 재귀적으로 처리
+ * - 자식 요소들에 대해 재귀적으로 `processVNode`를 호출하여 처리
+ *
+ * @param {VNode} vNode - 처리할 가상 DOM 노드
+ * @returns {VNodeChild} - 렌더링 가능한 형태로 변환된 노드
+ */
+function processVNode(vNode: VNode): VNodeChild {
+  if (vNode === null || vNode === undefined || typeof vNode === 'boolean') {
+    return '';
+  }
+
+  if (isVNodeText(vNode)) {
+    return String(vNode);
+  }
+
+  if (typeof vNode.type === 'function') {
+    return processVNode(vNode.type(vNode.props));
+  }
+
+  if (Array.isArray(vNode.children)) {
+    vNode.children = vNode.children.map(processVNode);
+  }
+
+  return vNode;
+}
+
+/**
+ * DOM 요소의 속성을 업데이트합니다.
+ *
+ * 이 함수는 다음을 수행합니다:
+ * - 이전 속성에서 제거된 속성을 처리합니다.
+ * - 새로운 속성에 대한 추가 및 업데이트를 처리합니다.
+ * - 이벤트 리스너, className, style 등 특별한 경우를 처리합니다.
+ *   - 이벤트 리스너는 'on'으로 시작하는 속성을 확인하여 처리합니다.
+ *   - 직접 `addEventListener`를 사용하지 않고, eventManager의 `addEvent` 및 `removeEvent` 함수를 사용합니다.
+ *   - 이는 이벤트 위임을 통해 효율적으로 이벤트를 관리하기 위함입니다.
+ *
+ * @param {HTMLElement} $element - 속성을 업데이트할 DOM 요소
+ * @param {VNodeProps} _oldProps - 이전 속성 집합
+ * @param {VNodeProps} _newProps - 새로운 속성 집합
+ */
+function updateAttributes(
+  $element: HTMLElement,
+  _oldProps: VNodeProps,
+  _newProps: VNodeProps
+) {
+  const oldProps = _oldProps ?? {};
+  const newProps = _newProps ?? {};
+
+  // 이전 속성 제거
+  for (const key in oldProps) {
+    if (!(key in newProps)) {
+      removeElementAttribute($element, key);
+    }
+  }
+
+  // 새 속성 추가 또는 업데이트
+  for (const key in newProps) {
+    const oldValue = oldProps?.[key];
+    const newValue = newProps[key];
+
+    if (newValue !== oldValue) {
+      setElementAttributes($element, { [key]: newValue });
+    }
+  }
```

Comment:
```suggestion
  const props = Object.entries(newProps).reduce((acc, [key, value]) => {
    return {
      ..acc,
      ...(oldProps?.[key] !== value && { [key]: value })
    }
  }, {});
  
  setElementAttributes($element, props);
```
이런식으로 표현할 수도 있답니다 ㅋㅋ